### PR TITLE
Build drivers only active 40

### PR DESCRIPTION
**Pull request by jeff303**
Cherry pick of #17262 for `release-x.40.x`

Also cherry pick #17101 for `release-x.40.x`.

Chicken and egg problem: this fix had failing tests on maria latest, the branch to cherry pick the maria fix had failing tests in building drivers. Just combine them and let's see all green.

### CODE REVIEW

**Review by dpsutton — APPROVED** [review]

**Comment by dpsutton** [comment]
I'm still a bit freaked out by this fix. What was left over in the modules directory? That to me is absolutely scary and makes this feel like a bandaid

**Comment by jeff303** [comment]
> I'm still a bit freaked out by this fix. What was left over in the modules directory? That to me is absolutely scary and makes this feel like a bandaid

There are a few scenarios I saw.  In CircleCI, the `src` and `test` directories were sticking around; see the output [here](https://app.circleci.com/pipelines/github/metabase/metabase/19427/workflows/b8640b74-9594-4c84-86d9-bfa28f8f5db9/jobs/799477/parallel-runs/0/steps/0-108).  Since `git` only tracks files (not directories), this kind of makes sense.  The executor is shared across multiple runs.  In the first run, a branch is built (perhaps master) that contains `presto-common`, then later on, it's used to build a `release-x.40.x` branch.  On that second run, it does a `git checkout` of the appropriate Metabase branch, but those directories are still there (since git does not delete newly empty directories if you switch branches). 

Locally, I also saw some generated files (ex: `target` directory) stick around, but I am constantly switching branches and rarely run `lein clean`, so that's probably why.  Also, as per conversation with @camsaul , we can't use `git` commands themselves to clear out these untracked files/dirs, because they are actually marked as ignored via the root-level `.gitignore` (on purpose, to prevent people from checking in generated or build artifacts).

So basically, the logic was changed such that the mere presence of a directory wasn't enough to be considered a full driver source package, but instead, a build file needs to be there too.

**Comment by dpsutton** [comment]
@jeff303 that makes sense. As soon as we are green go ahead and merge then. Thanks for explaining.

**Comment by jeff303** [comment]
There is now a particular BigQuery test that is consistently timing out (i.e. error).  I guess this needs dealt with before we can merge (or else need to override the check).

```
ERROR in (test-variance) (data.clj:170)

:bigquery
expected: (= 0.59 (test-aggregation [:var [:field (data/id :venues :price) nil]]))
  actual: clojure.lang.ExceptionInfo: Timed out after 60.0 s.
{:query {:database 43, :type :query, :query {:aggregation [[:var [:field 590 nil]]], :source-table 156}}}
```

**Comment by camsaul** [comment]
Something is going on with BigQuery (unrelated) but since this fixes other failures in CI we're going to merge anyway (discussed with @jeff303) so we can unblock other stuff

**Pull request merged**
